Added basic printing server.

mwhittaker opened the pull request:
I'm still getting the hang of communicating over the network in rust. Heck, I'm still getting used to rust. I'm starting off raft with a basic client/server. This implements a server that prints the messages line by line that it receives.

To test it out, you can use telnet. For example, in one shell run:

```
$ raft server
```

and in another shell run:

```
$ telnet localhost 9999
```

then type away.

mwhittaker committed:
Added basic printing server.

I'm still getting the hang of communicating over the network in rust.
Heck, I'm still getting used to rust. I'm starting off raft with a basic
client/server. This implements a server that prints the messages line by
line that it receives.

To test it out, you can use telnet. For example, in one shell run:

    $ raft server

and in another shell run:

    $ telnet localhost 9999

then type away.

mwhittaker commented:
@yuxiaotan @ambarsoni @alugupta

yuxiaotan commented on src/main.rs:
```diff
@@ -1,3 +1,24 @@
-    println!("Hello, world!");
+    let args = os::args();
+
+    if args.len() < 2 {
```
Please use static var to indicate why this is 2?

mwhittaker replied:
The first command line argument is always the name of the executable; here, it would be `raft`. If the number of arguments is less than two, then the executable was run with no command line arguments.

yuxiaotan replied:
Whoa weird. Sounds good to me!

yuxiaotan commented on src/server.rs:
```diff
@@ -0,0 +1,43 @@
+}
+
+fn exec() -> IoResult<()> {
+    let     listener = try!(TcpListener::bind("127.0.0.1", 9999u16));
```
Same as above

mwhittaker replied:
I'll make this static.

yuxiaotan replied:
Thanks!

yuxiaotan commented on src/main.rs:
```diff
@@ -1,3 +1,24 @@
+use std::os;
+
```
Unclear the purpose of this main. Is it client or server side?

mwhittaker replied:
Both. `raft client` will launch the client and `raft server` will launch the server.

yuxiaotan commented on src/main.rs:
```diff
@@ -1,3 +1,24 @@
+        return;
+    }
+
+    match args[1].as_slice() {
```
Why does one guy launch both server/client?

mwhittaker replied:
No good reason besides the fact that it's more convenient. It'll make less sense when raft is bigger than a handful of lines.

yuxiaotan replied:
Cool awesome.

mwhittaker commented:
Thanks for the comments @yuxiaotan! I'll incorporate your feedback when I get home :D

mwhittaker committed:
Made server's addres static.

This addresses comments made in [this pull
request](https://github.com/mwhittaker/raft/pull/4)
